src/ctia/stores/es/crud.clj

```diff
@@ -91,23 +93,43 @@
                     (build-create-result model coerce-fn)))
                 (remove-es-actions items) models)}))
 
-
-(s/defn get-docs-with-indices
+(defn get-docs-with-indices
   "Retrieves a documents from a search \"ids\" query. It enables to retrieves
  documents from an alias that points to multiple indices.
 It returns the documents with full hits meta data including the real index in which is stored the document."
-  [{:keys [conn index]} :- ESConnState
-   ids :- [s/Str]
+  [{:keys [conn index] :as _conn-state}
+   ids
    es-params]
   (let [ids-query (q/ids (map ensure-document-id ids))
         res (ductile.doc/query conn
                                index
                                ids-query
                                (assoc (make-es-read-params es-params)
+                                      :limit (count ids)
                                       :full-hits?
                                       true))]
     (:data res)))
 
+(defn lazy-get-docs-with-indices
+  "Retrieves a documents from a search \"ids\" query. It enables to retrieves
+ documents from an alias that points to multiple indices.
+It returns the documents with full hits meta data including the real index in which is stored the document."
+  ([conn-state ids es-params batch-size]
+   (lazy-get-docs-with-indices conn-state ids es-params batch-size get-docs-with-indices))
+  ([conn-state
+    ids
+    es-params
+    batch-size
+    docs-by-ids]
+   (when (seq ids)
```

Comment:
maybe `(s/pred seq)` the `ids` instead, this way one gets faster/clearer feedback than a nil result

Reply:
That's the end condition of the (lazy) recursion. 
BTW, if `ids` is also lazily generated (not likely but still), adding a schema would make  it eagerly consumed in order to check the schema. It's also probable that it could also break the lazyness of this function (I don't think so but it worths checking 😀, I will do so when I will be back from my ankle check).